{
  "schema_version": "1.6.3",
  "id": "UBUNTU-CVE-2021-46179",
  "details": "Reachable Assertion vulnerability in upx before 4.0.0 allows attackers to cause a denial of service via crafted file passed to the the readx function.",
  "aliases": [],
  "related": [
    "CVE-2021-46179"
  ],
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H"
    }
  ],
  "published": "2023-08-22T19:16:00Z",
  "modified": "2025-01-13T10:22:38Z",
  "affected": [
    {
      "package": {
        "ecosystem": "Ubuntu:Pro:16.04:LTS",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@3.91-1?arch=source&distro=esm-apps/xenial"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "3.91-1"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:Pro:18.04:LTS",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@3.94-4?arch=source&distro=esm-apps/bionic"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "3.94-2",
        "3.94-3",
        "3.94-4"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:20.04:LTS",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@3.95-2build1?arch=source&distro=focal"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "3.95-2",
        "3.95-2build1"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:22.04:LTS",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@3.96-3?arch=source&distro=jammy"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "3.96-2",
        "3.96-3"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:24.10",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@4.2.2-3?arch=source&distro=oracular"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "4.2.2-3"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:24.04:LTS",
        "name": "upx-ucl",
        "purl": "pkg:deb/ubuntu/upx-ucl@4.2.2-3?arch=source&distro=noble"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            }
          ]
        }
      ],
      "versions": [
        "3.96-3",
        "4.2.2-3"
      ],
      "ecosystem_specific": {
        "ubuntu_priority": "medium"
      }
    }
  ],
  "references": [
    {
      "type": "REPORT",
      "url": "https://ubuntu.com/security/CVE-2021-46179"
    },
    {
      "type": "REPORT",
      "url": "https://github.com/upx/upx/issues/545"
    },
    {
      "type": "REPORT",
      "url": "https://www.cve.org/CVERecord?id=CVE-2021-46179"
    }
  ]
}